tests/test_client.py

```diff
@@ -849,6 +851,210 @@ def test_download_all(mocker):
+
+    # first iteration in which job state is 'running' and two granules have completed
+    iter = client.iterator(status_page_json['jobID'], '/tmp')
+    granule_data = next(iter)
+    assert granule_data['bbox'] == BBox(-179.95,-89.95,179.95,89.95)
+    assert granule_data['path'].result() == '/tmp/fake.tif'
+
+    granule_data = next(iter)
+    assert granule_data['bbox'] == BBox(*extra_links[0]['bbox'])
+    assert granule_data['path'].result() == '/tmp/fake2.tif'
+
+    # job is still running and has another completed granule 
+    granule_data = next(iter)
+    assert granule_data['bbox'] == BBox(*extra_links[1]['bbox'])
+    assert granule_data['path'].result() == '/tmp/fake3.tif'
+
+    # job is paused and has another completed granule
+    granule_data = next(iter)
+    assert granule_data['bbox'] == BBox(*extra_links[2]['bbox'])
+    assert granule_data['path'].result() == '/tmp/fake4.tif'
+
+    # once the new granules are downloaded for the paused job the iterator stops
+    granule_data = next(iter, None)
+    assert granule_data == None
+
+    # need to start a new iterator after resuming
+    iter = client.iterator(status_page_json['jobID'], '/tmp')
+    # initial granules returned are the same as before (they are not re-downloaded by default)
+    granule_data = next(iter)
+    assert granule_data['bbox'] == BBox(-179.95,-89.95,179.95,89.95)
+    assert granule_data['path'].result() == '/tmp/fake.tif'
+    granule_data = next(iter)
+    assert granule_data['bbox'] == BBox(*extra_links[0]['bbox'])
+    assert granule_data['path'].result() == '/tmp/fake2.tif'
+    granule_data = next(iter)
+    assert granule_data['bbox'] == BBox(*extra_links[1]['bbox'])
+    assert granule_data['path'].result() == '/tmp/fake3.tif'
+    granule_data = next(iter)
+    assert granule_data['bbox'] == BBox(*extra_links[2]['bbox'])
+    assert granule_data['path'].result() == '/tmp/fake4.tif'
+    
+    # job is running and has another completed granule
+    granule_data = next(iter)
+    assert granule_data['bbox'] == BBox(*extra_links[3]['bbox'])
+    assert granule_data['path'].result() == '/tmp/fake5.tif'
+
+    # job has succeeded and has another completed granule
+    granule_data = next(iter)
+    assert granule_data['bbox'] == BBox(*extra_links[4]['bbox'])
+    assert granule_data['path'].result() == '/tmp/fake6.tif'
+
+    # iterator completes after job is in a final state and there are no new granules
+    granule_data = next(iter, None)
+    assert granule_data == None
+
+def test_iterator_failed_job(mocker):
+    failed_job = expected_job('C123', 'abc123')
+    failed_job['status'] = 'failed'
+    failed_job['message'] = 'Job failed'
+    get_json_mock = mocker.Mock(return_value = failed_job)
+    mocker.patch('harmony.harmony.Client._get_json', get_json_mock)
+    client = Client(should_validate_auth=False)
+
+    # first iteration in which job state is 'running' and two granules have completed
```

Comment:
I'm not following where the job starts out 'running' and two granules have completed. I think we should set the test up that way where a couple iterations work and then a third iteration the job is marked as failed and the iterator stops with an appropriate failure message, but I'm not sure where that's happening.

Reply:
That comment was cut and paste from above and does not belong here - removed.

Reply:
Can you make the test the way described - two successes and then third one the job status is failed and it exits out as expected? I just want to make sure we have a test that the iterator handles a change to the failed state when it didn't start out as failed.

Reply:
Done.